test/CDB_OAuth.sql

```diff
@@ -52,6 +56,8 @@ SELECT 'denied_tableas', catch_permission_error($$SELECT * FROM test_tablesas;$$
 SELECT 'denied_view', catch_permission_error($$SELECT * FROM test_view;$$);
 SELECT 'denied_mview', catch_permission_error($$SELECT * FROM test_mview;$$);
 SELECT 'denied_selectinto', catch_permission_error($$SELECT * FROM test_selectinto;$$);
+SELECT test_function();
+SELECT 'denied_function', catch_permission_error($$DROP FUNCTION test_function();$$);
```

Comment:
Could you leave a comment explaning why you are testing dropping the table instead of selecting it, plese? Something similar to https://github.com/CartoDB/cartodb-postgresql/pull/379#issuecomment-550487752

Reply:
In these tests we want to ensure that only the owner can do "privileged" actions, and that the owner is set to the right user during the oAuth flow. In the case of functions, the only "privileged" action is drop, so in order to test this behavior the steps are:

- create the function from the `creator_role` session
- perform public action (select) which should be allowed
- perform privileged action (drop) which should be allowed
- create the function again
- change session to `owner_role`
- perform public action (select) which should be allowed
- perform privileged action (drop) --> result will depend on the use case being tested (the trigger being enabled or not plus the role having the necessary permissions)

In the last test use case, the first drop attempted from the `creator_role` session is expected to fail since the trigger is enabled and well configured and the owner is then rightly set to `owner_role` and only the owner can do the drop.

Reply:
What I meant is, can you write it down with the tests so we know about it when we read it months from now?

Reply:
Oh! Sorry I misunderstood it. Of course!